src/Functions/randomFixedString.cpp

```diff
@@ -0,0 +1,90 @@
+#include <Columns/ColumnFixedString.h>
+#include <DataTypes/DataTypeFixedString.h>
+#include <Functions/FunctionFactory.h>
+#include <Functions/FunctionHelpers.h>
+#include <Functions/IFunctionImpl.h>
+#include <pcg_random.hpp>
+#include <Common/randomSeed.h>
+#include <common/arithmeticOverflow.h>
+#include <common/unaligned.h>
+
+#include <common/defines.h>
+
+namespace DB
+{
+namespace ErrorCodes
+{
+    extern const int ILLEGAL_TYPE_OF_ARGUMENT;
+    extern const int ILLEGAL_COLUMN;
+    extern const int DECIMAL_OVERFLOW;
+}
+
+
+/* Generate random fixed string with fully random bytes (including zero). */
+class FunctionRandomFixedString : public IFunction
+{
+public:
+    static constexpr auto name = "randomFixedString";
+
+    static FunctionPtr create(const Context &) { return std::make_shared<FunctionRandomFixedString>(); }
+
+    String getName() const override { return name; }
+
+    bool isVariadic() const override { return false; }
+
+    size_t getNumberOfArguments() const override { return 1; }
+
+    DataTypePtr getReturnTypeImpl(const ColumnsWithTypeAndName & arguments) const override
+    {
+        if (!isUnsignedInteger(arguments[0].type))
+            throw Exception("First argument for function " + getName() + " must be unsigned integer", ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT);
+
+        if (!arguments[0].column)
+            throw Exception("First argument for function " + getName() + " must be constant", ErrorCodes::ILLEGAL_COLUMN);
+
+        const size_t n = arguments[0].column->getUInt(0);
```

Comment:
We must ensure that the second argument is constant.

Reply:
Maybe `first argument`? I added check with function isColumnConst for it.